.github/workflows/ci.yaml

```diff
@@ -178,4 +178,78 @@ jobs:
           path: |
             tests/e2e/output/logs/
           retention-days: 7
+
+
+  ppnd-e2e-test:
```

Comment:
maybe we can do it as a follow up PR, but we don't want to copy and paste all of these steps. I see looking at the other jobs that we already had an issue there. There seems to be something called "Composite Actions" in Github Actions that maybe we can try.

Reply:
We may also want to do what Numaflow is doing, in which `case` is a matrix: https://github.com/numaproj/numaflow/blob/main/.github/workflows/ci.yaml#L205. Maybe we could have one matrixed job called "MultiStrategyE2E" and another called "SingleStrategyE2E" or something. The first would include our "functional-test" and any other future multi-strategy test. The second would include the ppnd test and any other single-strategy test.